packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx

```diff
@@ -253,84 +233,20 @@ const VisualizationCard: FC = () => {
-                                        <li>
-                                            <Link
-                                                target="_blank"
-                                                to={`/projects/${projectUuid}/dashboards/${dashboard.uuid}`}
-                                            >
-                                                {dashboard.name}
-                                            </Link>
-                                        </li>
-                                    );
-                                })}
-                            </ul>
-                        </>
-                    )}
-                </div>
-                <div className={Classes.DIALOG_FOOTER}>
-                    <div className={Classes.DIALOG_FOOTER_ACTIONS}>
-                        <Button
-                            disabled={isDeleting}
-                            onClick={() => setIsDeleteDialogOpen(false)}
-                        >
-                            Cancel
-                        </Button>
-                        <Button
-                            disabled={isDeleting}
-                            intent="danger"
-                            onClick={() => {
-                                /*
-                                Check the location of `goBack` after deleting a chart
-                                if we land on an unsaved chart,
-                                we go back further into an empty explore
-                                */
-                                history.listen((loc, action) => {
-                                    if (action === 'POP') {
-                                        if (loc.pathname.includes('/tables/')) {
-                                            history.push(
-                                                `/projects/${projectUuid}/tables`,
-                                            );
-                                        }
-                                    }
-                                });
-
-                                if (savedChart?.uuid) {
-                                    deleteData(savedChart.uuid);
-                                    history.push('/');
-                                }
-
-                                setIsDeleteDialogOpen(false);
-                            }}
-                        >
-                            Delete
-                        </Button>
-                    </div>
-                </div>
-            </Dialog>
+            {isDeleteDialogOpen && (
+                <DeleteActionModal
+                    isOpen={isDeleteDialogOpen}
+                    onClose={() => {
+                        return !isDeleting
+                            ? setIsDeleteDialogOpen(false)
+                            : undefined;
+                    }}
+                    uuid={savedChart?.uuid || ''}
+                    name={chartName || ''}
```

Comment:
When possible, is better to check the value in the render condition

`{isDeleteDialogOpen && savedChart?.uuid && (`